tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Parsing/CodeownersEntryPathComparer.cs

```diff
@@ -0,0 +1,260 @@
+                {
+                    return -1; // pathX is parent of pathY, so pathX comes first
+                }
+                if (normalizedPathX.StartsWith(normalizedPathY + "/", StringComparison.Ordinal))
+                {
+                    return 1; // pathY is parent of pathX, so pathY comes first
+                }
+            }
+
+            // Do alphabetical comparison of paths
+            int alphabeticalComparison = string.Compare(normalizedPathX, normalizedPathY, StringComparison.Ordinal);
+
+            // Handle wildcard precedence for similar paths
+            if (alphabeticalComparison != 0)
+            {
+                // Check for simple wildcard cases like "sdk/storage*" vs "sdk/storage"
+                if (HandleSimpleWildcardCase(x, y, normalizedPathX, normalizedPathY, out int wildcardResult))
+                {
+                    return wildcardResult;
+                }
+                
+                return alphabeticalComparison;
+            }
+
+            // If paths are identical alphabetically, prioritize wildcard paths
+            bool xHasWildcard = x.ContainsWildcard;
+            bool yHasWildcard = y.ContainsWildcard;
+
+            if (xHasWildcard && !yHasWildcard)
+            {
+                return -1; // x (with wildcard) comes before y (without wildcard)
+            }
+            if (!xHasWildcard && yHasWildcard)
+            {
+                return 1; // y (with wildcard) comes before x (without wildcard)
+            }
+
+            // If both have wildcards or both don't have wildcards, they're equal
+            return 0;
+        }
+
+        /// <summary>
+        /// Handle simple wildcard cases where one path ends with * and matches the other path.
+        /// </summary>
+        private bool HandleSimpleWildcardCase(CodeownersEntry x, CodeownersEntry y, string pathX, string pathY, out int result)
+        {
+            result = 0;
+            bool xHasWildcard = x.ContainsWildcard;
+            bool yHasWildcard = y.ContainsWildcard;
+
+            // Simple check: if one path ends with * and matches the other path, prioritize the wildcard
+            if (xHasWildcard && !yHasWildcard && pathX.EndsWith("*") && pathX.TrimEnd('*') == pathY)
+            {
+                result = -1; // x (with wildcard) comes before y (without wildcard)
+                return true;
+            }
+            if (!xHasWildcard && yHasWildcard && pathY.EndsWith("*") && pathY.TrimEnd('*') == pathX)
+            {
+                result = 1; // y (with wildcard) comes before x (without wildcard)
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Compare two CodeownersEntry objects by their service labels.
+        /// </summary>
+        private int CompareByServiceLabel(CodeownersEntry x, CodeownersEntry y)
```

Comment:
This can be static if it doesn't depend on anything in a CodeownersEntryPathComparer instance.